src/modules/aptextrapackages/ExtraPackagesViewStep.cpp

```diff
@@ -0,0 +1,137 @@
+/*******************************************************************************
+ * Copyright (c) 2016-2017, Kyle Robbertze (AIMS, South Africa)
+ *
+ * This project is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * It is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this project. If not, see <http://www.gnu.org/licenses/>.
+ *
+ *    Description:  Module for installing select packages in the installed
+ *                  system
+ *
+ *        Created:  06/12/2016 07:31:14
+ *       Compiler:  g++
+ *
+ *         Author:  Kyle Robbertze (kr), kyle@aims.ac.za
+ *******************************************************************************/
+
+#include "ExtraPackagesViewStep.h"
+
+#include "ExtraPackagesPage.h"
+#include "utils/Logger.h"
+#include "JobQueue.h"
+
+#include <QVariant>
+
+CALAMARES_PLUGIN_FACTORY_DEFINITION( ExtraPackagesViewStepFactory,
+                                     registerPlugin<ExtraPackagesViewStep>(); )
+
+ExtraPackagesViewStep::ExtraPackagesViewStep( QObject* parent ) :
+    Calamares::ViewStep( parent )
+{
+    emit nextStatusChanged( true );
+    m_widget = new ExtraPackagesPage();
+}
+
+ExtraPackagesViewStep::~ExtraPackagesViewStep()
+{
+    if ( m_widget && m_widget->parent() == nullptr )
+        m_widget->deleteLater();
+}
+
+QString ExtraPackagesViewStep::prettyName() const
+{
+    return tr( "Extra Packages" );
+}
+
+QString ExtraPackagesViewStep::prettyStatus() const
+{
+    return m_widget->prettyStatus();
+}
+
+QWidget* ExtraPackagesViewStep::widget()
+{
+    return m_widget;
+}
+
+void ExtraPackagesViewStep::next()
+{
+    m_widget->checkInternet();
```

Comment:
`onActivate` is called when Calamares is started. We think users will see the note about needing internet, then go and plug in/connect to wifi. This means they simply need to go back and forward to enable the module after connecting instead of having to restart the installer